airbyte-cdk/python/unit_tests/sources/streams/http/test_http.py

```diff
@@ -153,3 +154,73 @@ def test_stub_custom_backoff_http_stream(mocker):
         list(stream.read_records(SyncMode.full_refresh))
 
     # TODO(davin): Figure out how to assert calls.
+
+
+class PostHttpStream(StubBasicReadHttpStream):
+    http_method = "POST"
+
+    def parse_response(self, response: requests.Response, **kwargs) -> Iterable[Mapping]:
+        """Returns response data as is"""
+        yield response.json()
+
+
+class TestRequestBody:
+    """Suite of different tests for request bodies"""
+
+    json_body = {"key": "value"}
+    data_body = "key:value"
+
+    def request2response(self, request, context):
+        return json.dumps({"body": request.text, "content_type": request.headers.get("Content-Type")})
+
+    def test_json_body(self, mocker, requests_mock):
+
+        stream = PostHttpStream()
+        mocker.patch.object(stream, "request_body_json", return_value=self.json_body)
+
+        requests_mock.register_uri("POST", stream.url_base, text=self.request2response)
+        response = list(stream.read_records(sync_mode=SyncMode.full_refresh))[0]
+
+        assert response["content_type"] == "application/json"
+        assert json.loads(response["body"]) == self.json_body
+
+    def test_text_body(self, mocker, requests_mock):
```

Comment:
we should also have a test which verifies that if a dict is returned from `request_body_data` then header is `application/x-www-form-urlencoded` and the body is `key=value`

Reply:
Added